File: frontend/src/services/learningResourceService.ts
import api from '../utils/axiosConfig';
import { LearningResource } from '../types/learningPlan';

export const learningResourceService = {
    // Get all resources for a learning plan
    getPlanResources: async (planId: number): Promise<LearningResource[]> => {
        const response = await api.get(`/learning-resources/plans/${planId}`);
        return response.data;
    },

    // Get a specific resource by ID
    getResourceById: async (id: number): Promise<LearningResource> => {
        const response = await api.get(`/learning-resources/${id}`);
        return response.data;
    },

    // Create a new resource
    createResource: async (planId: number, resourceData: Omit<LearningResource, 'id' | 'completed'>): Promise<LearningResource> => {
        const response = await api.post(`/learning-resources/plans/${planId}`, resourceData);
        return response.data;
    },

    // Update an existing resource
    updateResource: async (id: number, resourceData: Partial<LearningResource>): Promise<LearningResource> => {
        const response = await api.put(`/learning-resources/${id}`, resourceData);
        return response.data;
    },

    // Delete a resource
    deleteResource: async (id: number): Promise<void> => {
        await api.delete(`/learning-resources/${id}`);
    }
}; 

File: backend/src/main/java/com/paf/chop/backend/services/LearningResourceService.java
package com.paf.chop.backend.services;

import com.paf.chop.backend.dto.request.LearningResourceRequestDTO;
import com.paf.chop.backend.dto.response.LearningResourceResponseDTO;
import java.util.List;

public interface LearningResourceService {
    
    /**
     * Creates a new learning resource
     * 
     * @param requestDTO the learning resource details
     * @param planId the ID of the learning plan
     * @param userId the ID of the user creating the resource
     * @return the created learning resource response
     */
    LearningResourceResponseDTO createResource(LearningResourceRequestDTO requestDTO, Long planId, Long userId);
    
    /**
     * Updates an existing learning resource
     * 
     * @param id the ID of the resource to update
     * @param requestDTO the updated resource details
     * @param userId the ID of the user updating the resource
     * @return the updated learning resource response
     */
    LearningResourceResponseDTO updateResource(Long id, LearningResourceRequestDTO requestDTO, Long userId);
    
    /**
     * Deletes a learning resource
     * 
     * @param id the ID of the resource to delete
     * @param userId the ID of the user deleting the resource
     */
    void deleteResource(Long id, Long userId);
    
    /**
     * Gets a learning resource by ID
     * 
     * @param id the ID of the resource to retrieve
     * @param userId the ID of the user requesting the resource
     * @return the learning resource response
     */
    LearningResourceResponseDTO getResourceById(Long id, Long userId);
    
    /**
     * Gets all resources for a learning plan
     * 
     * @param planId the ID of the learning plan
     * @param userId the ID of the user requesting the resources
     * @return list of learning resource responses
     */
    List<LearningResourceResponseDTO> getPlanResources(Long planId, Long userId);
    
    /**
     * Gets all resources for a public learning plan
     * 
     * @param planId the ID of the public learning plan
     * @return list of learning resource responses
     */
    List<LearningResourceResponseDTO> getPublicPlanResources(Long planId);
} 